Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLReduction.py

```diff
@@ -545,6 +547,46 @@ def _apply_masks(self, ws):
         if mask_ws:
             self._mask(ws, mask_ws)
 
+    def _apply_thickness(self, ws):
+        """
+            Perform the normalization by sample thickness. In case the provided
+            thickness is -1, this method will try to get it from the sample
+            logs.
+            @param ws : input workspace on wich the normalization is applied.
+        """
+        thickness = self.getProperty('SampleThickness').value
+        if thickness == -1:
+            try:
+                run = mtd[ws].getRun()
+                thickness = run.getLogData('sample.thickness').value
```

Comment:
When using the value from the nexus file, can you log an information what that value is?

Reply:
Done